testing/python/fixtures.py

```diff
@@ -1312,6 +1312,36 @@ def test2(no_eq):
         result = pytester.runpytest()
         result.stdout.fnmatch_lines(["*4 passed*"])
 
+    def test_reorder_indirect(self, pytester: Pytester) -> None:
```

Comment:
A docstring with a reference to the PR (since there is no issue) would be good.

BTW, if you explicitly want to test the reordering here and not the runtime behavior, I suggest running the pytester with `--collect-only --quiet` which just spits the test order, and checking that, instead of doing prints and all that.

Reply:
> explicitly want to test the reordering here

Afaik pytest doesn't give guarantees about absolute test order, but only about certain constraints. So I intended to assert for the number of setups/teardowns to be minimal, not for a specific sequence. I think that's only possible if we actually let the fixture requests happen. They happen after collect.

But `--collect-only` could still be good for kind of unit testing the reorder function, would such an additional test be useful?